src/views/tutorials-landing/components/product-section/index.tsx

```diff
@@ -0,0 +1,85 @@
+import { IconArrowRight24 } from '@hashicorp/flight-icons/svg-react/arrow-right-24'
+import ProductIcon from 'components/product-icon'
+import StandaloneLink from 'components/standalone-link'
+import s from './product-section.module.css'
+import CollectionContentCardLink from 'components/tutorials-landing-view/collection-content-card-link'
+import CertificationContentCardLink from 'components/tutorials-landing-view/certification-content-card-link'
+
+const ProductSection = ({
+	certification,
+	featuredCollections,
+	featuredUseCases,
+	product,
+}: $TSFixMe) => {
```

Comment:
I think my feedback here is largely that we should first implement this component with a view model first, rather than passing the domain models in.

A very rough example of this component with a View Model approach, to illustrate:

```typescript
interface ProductSectionProps {
	id: string
	logo: ReactElement
	description: string
	topLinks: Array<StandaloneLinkProps>
	cards: Array<ReactElement>
	extraLinks?: ExtraLinks
}

interface ExtraLinks {
	title: string
	links: Array<StandaloneLinkProps>
}
```

Reply:
@BrandonRomano Since `ProductSection` is a composition of many base UI components and it's only needed on this page in one form, I'm not convinced we need to abstract the extra view model layer in this case. Can you share a little more about value you think this abstraction will give us?

Reply:
So, largely the motivation for wanting to implement the view model has to do with future proofing.  Future proofing in both the "this may be used somewhere else in some other context", and also in the "Our content desires might be adjusted / extended in the future".

I'll provide a few examples for the _later_, because this page evolving over time is much more of a real thing.
- We may in the future want to have cards that don't just link to collections.  Maybe the "Collections of Collections" idea manifests in the future, and we want to render a different DomainModel implementation in the cards list.
- I could very easily imagine the links of "Featured use cases" in the future we want to leverage linking to something else than use cases.
- We may simply just want to render this presentation in a section below to communicate a use-case rather than a product.

These scenarios are not destined to happen, but in explicitly decoupling the _domain_ (e.g. collection) from the presentation, these possibilities become easier to support, as the View Layer would happy render any future domain that our future business needs will require.

Reply:
@BrandonRomano thank you for the elaboration! Going to continue chatting via Slack to save a little time 😊

Reply:
Sharing an update from Slack for posterity: we will make non-release-blocking Asana tasks for following up on abstractions